### PR TITLE
Use Signal language for media preview timestamp

### LINKED ISSUE #3449: Time/date not translated in image view
2.19.0
_Edit: see third comment for a proper bug report._
phil thomas katt enterprise room

### PR DESCRIPTION
**Pull request by FeuRenard**
### First time contributor checklist
- [x] I have read [how to contribute](https://github.com/WhisperSystems/Signal-Android/blob/master/CONTRIBUTING.md) to this project
- [x] I have signed the [Contributor Licence Agreement](https://whispersystems.org/cla/)
### Contributor checklist
- [x] I am following the [Code Style Guidelines](https://github.com/WhisperSystems/Signal-Android/wiki/Code-Style-Guidelines)
- [x] I have tested my contribution on these devices:
  - Sony Xperia U, Android 4.4.4 (CyanogenMod)
- [x] My contribution is fully baked and ready to be merged as is
- [x] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in my commit message

---
### Description

Fixes #3449 

Before this PR Android's `DateUtils` is used to get the timespan string in `MediaPreviewActivity`. This class' methods have no parameters to provide a locale. So I had to write our own method to get the timespan. This method will respect the language set inside Signal.

We also could use the same timespan string like in the message bubbles or the conversation list. But as there is enough space in the media preview action bar subtitle, I decided to keep the unabbreviated format (e.g. "Monday" instead of "Mon" or "minute" instead of "min"). This required writing the new method.

// FREEBIE

### CODE REVIEW

**Comment by moxie0** [comment]
Thanks but please resubmit without changes to the translations

**Comment by FeuRenard** [comment]
I dropped the two commits that renamed and moved around strings.

**Comment by moxie0** [comment]
What's wrong with the existing methods we use to calculate relative time strings?

**Comment by FeuRenard** [comment]
Android's methods don't respect Signal's internal language.
Signal's existing methods only deliver abbreviated relative time span strings ("Mon" instead of "Monday" or "min" instead of "minute"). Before this PR (current master) the unabbreviated forms are displayed in the media preview action bar (but with system language instead of Signal language). As the media preview offers enough space for the unabbreviated forms I decided to stick to them.

**Comment by moxie0** [comment]
Let's stick with the existing abbreviated strings

**Comment by FeuRenard** [comment]
Now the same string as in the message bubbles is used.

File: src/org/thoughtcrime/securesms/util/DateUtils.java [hunk 1]
```diff
@@ -68,7 +68,7 @@ public static String getExtendedRelativeTimeSpanString(final Context c, final Lo
     if (isWithin(timestamp, 1, TimeUnit.MINUTES)) {
       return c.getString(R.string.DateUtils_now);
     } else if (isWithin(timestamp, 1, TimeUnit.HOURS)) {
-      int mins = (int)TimeUnit.MINUTES.convert(System.currentTimeMillis() - timestamp, TimeUnit.MILLISECONDS);
+      int mins = convertDelta(timestamp, TimeUnit.MINUTES);
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/util/DateUtils.java]: is this change necessary?

Comment (FeuRenard) [inline src/org/thoughtcrime/securesms/util/DateUtils.java]: No. You can just drop https://github.com/WhisperSystems/Signal-Android/pull/5600/commits/388aeaec84d195cc8d69bb003c7a5a6f142c1888 while squashing the commits. It improves readability though.

**Comment by FeuRenard** [comment]
Rebased and removed unnecessary refactoring.

**Comment by moxie0** [comment]
in 3.24.0
